frontend/javascripts/oxalis/view/right-border-tabs/trees_tab/tree_hierarchy_view.tsx

```diff
@@ -264,6 +268,27 @@ function TreeHierarchyView(props: Props) {
     ? [getNodeKey(GroupTypeEnum.GROUP, props.activeGroupId)]
     : props.selectedTreeIds.map((treeId) => getNodeKey(GroupTypeEnum.TREE, treeId));
 
+  // biome-ignore lint/correctness/useExhaustiveDependencies: The other dependencies change too often, they were omitted due to performance reasons
+  useEffect(() => {
+    // maybe expand group of the active tree
+    if (props.activeTreeId == null) return;
+    const sourceNode = props.trees[props.activeTreeId];
+    if (sourceNode.groupId == null) return; // tree is a direct child of the root group which is always expanded
+    const expandedGroups = additionallyExpandGroup(props.treeGroups, sourceNode.groupId, (id) =>
+      getNodeKey(GroupTypeEnum.GROUP, id),
+    );
+    if (expandedGroups == null) return;
+    // props.treeGroups are used to get the newly expanded groups but can be outdated, so the union needs to be used here
+    setExpandedGroups(new Set([...expandedNodeKeys, ...Array.from(expandedGroups)]));
```

Comment:
a) I don't understand the comment, because `expandedGroups` should contain the "newly expanded groups" (and not `expandedNodeKeys`) and b) I don't understand why the concatenation is needed in the first place? The implementation of `additionallyExpandGroup` seems to already contain all other previously expanded groups?

I would have expected simply:
```suggestion
    setExpandedGroups(expandedGroups);
```

or doesn't this work?

Reply:
you are right. previously I was setting the `expandedNodeKeys` directly, before understanding that I shouldnt mutate them directly. It used to be the right solution then but I didnt question it again after mutating the expanded nodes by dispatching an action.